### PR TITLE
feat(ob-sql-parser): upgrade antlr g4 for oceanbase v4.3.0

### PR DESCRIPTION
**Pull request by yizhouxw**
init ob 430 antlr g4

### CODE REVIEW

File: libs/ob-sql-parser/src/test/java/com/oceanbase/tools/sqlparser/adapter/OraclePartitionFactoryTest.java [hunk 1]
```diff
@@ -599,23 +599,9 @@ public void generate_listPartitionWithSubPartitionOptionsQuantity_succeed() {
         Assert.assertEquals(expect, actual);
     }
 
-    @Test
-    public void generate_columnPartition_succeed() {
```
Comment (yizhouxw) [inline libs/ob-sql-parser/src/test/java/com/oceanbase/tools/sqlparser/adapter/OraclePartitionFactoryTest.java:603]: there is no 'column partition' indeed, the case is for cover previous g4 syntax definition

File: libs/ob-sql-parser/src/main/resources/obmysql/sql/OBLexer.g4 [hunk 2]
```diff
@@ -19,6 +19,17 @@ lexer grammar OBLexer;
 public boolean inRangeOperator = false;
 }
 
+// Fragment rules for sub-patterns
+fragment NOTASCII : '\u0080'..'\u00FF';
```
Comment (yhilmare) [inline libs/ob-sql-parser/src/main/resources/obmysql/sql/OBLexer.g4:23]: these fragment tokens seems like useless~

File: libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4 [hunk 3]
```diff
@@ -1150,11 +1155,11 @@ alter_keystore_stmt
...
-    | CREATE temporary_option TABLE relation_factor LeftParen table_element_list RightParen table_option_list? opt_partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor table_option_list opt_partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor on_commit_option AS subquery order_by? fetch_next_clause?
+    : CREATE special_table_type TABLE relation_factor LeftParen table_element_list RightParen table_option_list? (partition_option | auto_partition_option)? (WITH_COLUMN_GROUP LeftParen column_group_list RightParen)? on_commit_option
+    | CREATE special_table_type TABLE relation_factor LeftParen table_element_list RightParen table_option_list? (partition_option | auto_partition_option)? (WITH_COLUMN_GROUP LeftParen column_group_list RightParen)? on_commit_option AS subquery order_by? fetch_next_clause?
```
[the remark this answers is not in the document]
Comment (yizhouxw) [reply libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4]: partition_option still exists

File: libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4 [hunk 4]
```diff
@@ -267,7 +270,9 @@ conf_const
 bool_pri
     : bit_expr IS not? (NULLX|is_nan_inf_value)
     | bit_expr IS NOT? (JSON FORMAT)? is_json_constrain
-    | bit_expr ((((COMP_EQ|COMP_GE)|(COMP_LE|COMP_LT)) SOME|((COMP_GT|COMP_NE) SOME|COMP_NE_PL))|(((COMP_GE|COMP_GT? COMP_EQ)|(COMP_LE|COMP_LT COMP_EQ?))|((COMP_LT? COMP_GT|COMP_NE)|(Caret|Not) COMP_EQ)) sub_query_flag?) bit_expr
+    | json_exists_expr
```
[the remark this answers is not in the document]
Comment (yizhouxw) [reply libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4:273]: seems not conflict ?

File: libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4 [hunk 5]
```diff
@@ -1150,11 +1155,11 @@ alter_keystore_stmt
     ;
 
 create_table_stmt
-    : CREATE temporary_option TABLE relation_factor LeftParen table_element_list RightParen table_option_list? opt_partition_option on_commit_option
-    | CREATE temporary_option TABLE relation_factor LeftParen table_element_list RightParen table_option_list? opt_partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor table_option_list opt_partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor partition_option on_commit_option AS subquery order_by? fetch_next_clause?
-    | CREATE temporary_option TABLE relation_factor on_commit_option AS subquery order_by? fetch_next_clause?
+    : CREATE special_table_type TABLE relation_factor LeftParen table_element_list RightParen table_option_list? (partition_option | auto_partition_option)? (WITH_COLUMN_GROUP LeftParen column_group_list RightParen)? on_commit_option
```
[the remark this answers is not in the document]
Comment (yizhouxw) [reply libs/ob-sql-parser/src/main/resources/oboracle/sql/OBParser.g4]: defined with_column_group now

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**